### PR TITLE
Added new solution for Project Euler problem 027

### PR DESCRIPTION
**Pull request by fpringle**
### **Describe your change:**

 Project Euler Problem 27: Quadratic primes

Problem Statement:

Euler discovered the remarkable quadratic formula:

n² + n + 41

It turns out that the formula will produce 40 primes for the consecutive values n = 0 to 39. However, when n = 40, 40² + 40 + 41 = 40(40 + 1) + 41 is divisible by 41, and certainly when n = 41, 41² + 41 + 41 is clearly divisible by 41.

The incredible formula  n2 − 79n + 1601 was discovered, which produces 80 primes for the consecutive values n = 0 to 79. The product of the coefficients, −79 and 1601, is −126479.

Considering quadratics of the form:
n² + an + b, where |a| < 1000 and |b| <= 1000 where |n| is the modulus/absolute value of ne.g. |11| = 11 and |−4| = 4

Find the product of the coefficients, a and b, for the quadratic expression that produces the maximum number of primes for consecutive values of n, starting with n = 0.

Solution: This is almost the same algorithm as sol1.py authored by @cclauss. The main difference is that we generate the primes at the beginning rather than checking for primeness on the fly. since for n=b, n^2 + a*n + b will always be divisible by b, the theoretical maximum number we have to check for primeness is (b_limit - 1)^2 + a_limit * b_limit + b_limit. For the real problem, a_limit = b_limit = 1000 so we need to generate primes up to 1998000, just under 2m. In practice, the largest prime we actually end up checking for primeness is 12989, but I can't see any way we could have determined that before running the algorithm. If there was, we'd only need to generate primes up to 13000 instead of 2000000, which would result in a code speedup of about 50%

Reference:  [https://projecteuler.net/problem=27](url)

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_027/sol2.py
Comment (xcodz-dot) [inline project_euler/problem_027/sol2.py:40]: If possible could you please add a doctest here

Comment (fpringle) [reply project_euler/problem_027/sol2.py]: Thanks, done

**Review by xcodz-dot — APPROVED** [review]

**Comment by xcodz-dot** [comment]
Now you will have to wait for a member approval or changes

**Comment by dhruvmanila** [comment]
Thank you for your contribution but currently, we are not accepting any solution to existing problems. Take a look at the `project_euler` directory and submit a solution to non-existing ones.

**Pull request closed without merging**
